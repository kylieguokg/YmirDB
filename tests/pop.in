pop a
set a 1 2 3
get a
pop a
pop a
pop a
pop a
get a
set b a 1
set c b 1
set d c 1
set e d 1
list entries
forward a
forward b
forward c
forward d
forward e
backward a
backward b
backward c
backward d
backward e
pop c
list entries
forward a
forward b
forward c
forward d
forward e
backward a
backward b
backward c
backward d
backward e
bye
