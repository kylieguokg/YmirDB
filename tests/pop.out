> no such key

> ok

> [1 2 3]

> 1

> 2

> 3

> nil

> []

> ok

> ok

> ok

> ok

> e [d 1]
d [c 1]
c [b 1]
b [a 1]
a []

> nil

> a

> a, b

> a, b, c

> a, b, c, d

> b, c, d, e

> c, d, e

> d, e

> e

> nil

> b

> e [d 1]
d [c 1]
c [1]
b [a 1]
a []

> nil

> a

> nil

> c

> c, d

> b

> nil

> d, e

> e

> nil

> bye
